### PR TITLE
Assorted minor fixes while testing new faas-cli subcommands

### PR DESCRIPTION
**Pull request by johnmccabe**
## Description

This PR includes a number of minor fixes to issues found when reviewing the new `faas-cli` subcommands - `invoke`, `list`, `new`.

Please see the individual commits (1 per issue) for more details on each issue.

- Fixes capitalisation in `invoke` help
- Fix handling for gateway URLs, removing trailing `\`
- Fix `url-ping` function so that it works when used with `invoke`
- Suppress `Reading from STDIN..` message when STDIN is piped to the `invoke`
- Fix handling of the default Gateway URL in the `invoke` command
- Fix help text in the `new` function command

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Tested locally on a swarm instance, details of the tested commands are given in each commit.jj

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: commands/invoke.go [hunk 1]
```diff
@@ -71,7 +71,11 @@ func runInvoke(cmd *cobra.Command, args []string) {
 		gatewayAddress = gateway
 	}
 
-	fmt.Fprintf(os.Stderr, "Reading from STDIN - hit (Control + D) to stop.\n")
+	stat, _ := os.Stdin.Stat()
+	if (stat.Mode() & os.ModeCharDevice) != 0 {
+		fmt.Fprintf(os.Stderr, "Reading from STDIN - hit (Control + D) to stop.\n")
```
Comment (ems5311) [inline commands/invoke.go]: Why to Stderr, not Stdout? Won't this cause TravisCi to break?

Comment (johnmccabe) [reply commands/invoke.go]: Nope, stderr won't impact travis - that responds to the exit code from the command.

You use stderr for informational messages that you don't want to impact the use of the command when chained together with other commands, for example piping the output from this to another command would only see the output written to stdout passed to the next command.

**Comment by ems5311** [comment]
When the `faas-cli list` or `faas-cli invoke` commands are issued, it looks like when the `-f` or `--yaml` flag is passed in, only the gateway URL is taken from the YAML file.

Would it be okay to list out the only the functions specified in the given YAML file that **are deployed** when you use `faas-cli list -f <filename>`? Similarly, only invoke the deployed functions specified in the given YAML file with `faas-cli invoke -f <filename>`?

Also, it looks like the `services` variable in both `list.go` and `invoke.go` are unused after being set by the parsed YAML file output.

File: commands/invoke.go [hunk 2]
```diff
@@ -63,22 +62,22 @@ func runInvoke(cmd *cobra.Command, args []string) {
...
 		if parsedServices != nil {
 			services = *parsedServices
-			gatewayAddress = services.Provider.GatewayURL
+			gateway = services.Provider.GatewayURL
 		}
 	}
 
-	if len(gateway) > 0 && gateway != "http://localhost:8080" {
-		gatewayAddress = gateway
+	stat, _ := os.Stdin.Stat()
```
Comment (alexellis) [inline commands/invoke.go]: Nice trick 👍

File: sample/url-ping/handler.py [hunk 3]
```diff
@@ -8,11 +8,12 @@ def print_url(url):
         print("Timed out trying to reach URL.")
 
 def handle(req):
-    print("Handle this -> " + req)
-    if req.find("http") == -1:
+    url = req.rstrip()
```
Comment (alexellis) [inline sample/url-ping/handler.py]: This is a good idea. You can also invoke this with echo -n "https://www.openfaas.com".

**Comment by alexellis** [comment]
Absolute legend 💯

**Comment by alexellis** [comment]
@ems5311 I think that issue is fixed  with the above. I'll cut an alpha release you can test out.

**Pull request merged**